lld/ELF/Driver.cpp

```diff
@@ -1436,6 +1436,8 @@ static void readConfigs(opt::InputArgList &args) {
   config->zInterpose = hasZOption(args, "interpose");
   config->zKeepTextSectionPrefix = getZFlag(
       args, "keep-text-section-prefix", "nokeep-text-section-prefix", false);
+  config->zLrodataAfterBss =
+      getZFlag(args, "lrodata-after-bss", "nolrodata-after-bss", false);
```

Comment:
Use `!config->isPic` as default value, instead of false? (moving below the assignment of `isPic`, of course)

Reply:
Sorry, but no. I don't think the x86-64 -fno-pic case justifies changing the tradition that -no-pie/-pie does not change the layout (internal linker script). `-z lrodata-after-bss` is a common ground that we both can accept and it has its own merit to test two layouts for file-backed transparent huge pages.

Reply:
OK...then it ought to default to true, instead of false.

(And I do also wish you would explain better why you feel so strongly about not changing behavior based on pic vs not.)

Reply:
default to true makes sense, it works for more cases

Reply:
I have explained. The layout should not magically change if the user switches between -no-pie and -pie/-shared. There are a few options to change the layout (-z keep-text-section-prefix, -z separate-code, --no-rosegment, etc), but none does a "smart" choice. 

With -z separate-code and a large max-page-size, one extra PT_LOAD can have a significant waste. I was actually struggling with the default but now I feel that the current layout makes more sense as the default.

Reply:
> With -z separate-code and a large max-page-size, one extra PT_LOAD can have a significant waste.

This makes sense for the more common `-pie` case.

But *everybody* who's linking with `-no-pie` on x86-64 and has large sections is going to need `.lrodata` at the end of the binary so it doesn't affect small data relocations. This basically forces the user to pass `-z lrodata-after-bss` if they're passing `-no-pie`, which seems like a pointless burden on the user. I don't think people are generally concerned about exactly where `.lrodata` is in the binary, the whole point is that it's *somewhere* far away, doesn't really matter where.

Reply:
I think using "everybody" is not accurate. I've mentioned that `-fpie -no-pie` does not need this. And this `-fno-pie -no-pie` use has been a tiny portion of the user community. We are, after all, dealing with a problem that very few users run into in the first place. I think it matters to optimize for the prevailing configurations (-fpie/-fpic with -pie).

We also need evidence that "I have heard that x86-64 -fno-pic is measurably slower than -fpie in large workloads"

Reply:
Thanks for understanding. I think I am fairly strong in keeping the stance that layout decisions should not magically change due to -no-pie/-pie. If we ever decide to deviate from the policy, we can re-consider. But having `-z lrodata-after-bss` should make these possible.

Reply:
When you say "policy" do you mean that the layout policy is documented somewhere?